### ISSUE TITLE
Extend putObject API to accept user metadata for the object

### ISSUE THREAD

**Issue by kishoresanke** [minio/minio-js#615]
Extend putObject API to accept user metadata for the object

**Comment by harshavardhana** [comment]
Thanks @kishoresanke we will be fixing this soon.

**Comment by harshavardhana** [comment]
This requires an API change to be implemented along the same lines as minio-go API in this regard https://github.com/minio/minio-go/issues/747

**Comment by ebozduman** [comment]
We need to finish the on-going minio-go put object refactoring effort before resolving this issue. 
#747

**Comment by deekoder** [comment]
We will do it in the future milestone. We are already on our way to having this in minio-go

**Comment by harshavardhana** [comment]
Unblocking since minio/minio-go#747 is accepted and released.

**Comment by deekoder** [comment]
Duplicate of #555

**Comment by deekoder** [comment]
Closing this issue here. @kishoresanke  please follow #555 for this request.
